File: src/main/java/tw/racket/lang/psi/RacketFile.java
package tw.racket.lang.psi;

import com.intellij.extapi.psi.PsiFileBase;
import com.intellij.openapi.fileTypes.FileType;
import com.intellij.psi.FileViewProvider;
import org.jetbrains.annotations.NotNull;
import tw.racket.lang.RacketFileType;
import tw.racket.lang.RacketLanguage;

public class RacketFile extends PsiFileBase {
    public RacketFile(@NotNull FileViewProvider viewProvider) {
        super(viewProvider, RacketLanguage.INSTANCE);
    }

    @NotNull
    @Override
    public FileType getFileType() {
        return RacketFileType.INSTANCE;
    }

    @Override
    public String toString() {
        return "Racket File";
    }
}

File: src/main/kotlin/tw/racket/lang/psi/RacketFile.kt
package tw.racket.lang.psi

import com.intellij.extapi.psi.PsiFileBase
import com.intellij.openapi.fileTypes.FileType
import com.intellij.psi.FileViewProvider
import tw.racket.lang.RacketLanguage
import tw.racket.lang.RacketFileType

class RacketFile(viewProvider: FileViewProvider) : PsiFileBase(viewProvider, RacketLanguage) {
    override fun getFileType(): FileType {
        return RacketFileType
    }

    override fun toString(): String {
        return "Racket File"
    }
}